Design Picker: Make blank canvas as button

arthur791004 opened the pull request:
#### Changes proposed in this Pull Request

* As I mentioned [here](https://github.com/Automattic/wp-calypso/issues/58429#issuecomment-978813504), I think we don't need to do A/B test for this change, I use config named `design-picker-use-featured-picks-buttons` to make the design with “featured” term in the theme_picks show at the footer of `<DesignPickerCategoryFilter />`. If we find the drop of step conversion or flow completion, we can easily turn off this feature.
* Update the `.stylelintrc` to enable general styles: `font-weight: 500;` and `border-radius: 4px;`

![image](https://user-images.githubusercontent.com/13596067/143820893-dcefb3dc-4ef3-41b3-b839-af21ac62e38f.png)

![image](https://user-images.githubusercontent.com/13596067/143820916-45407b9a-7c9f-4e53-b071-bcf1032289ae.png)


#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

**Hero Flow**

* Go to `/start/setup-site?siteSlug=<your_site>`
* Select Build intent
* Check the blank canvas button show below the categories

**Do it for me**

* Go to `/start/do-it-for-me?siteSlug=<your_site>`
* Check the design picker without blank canvas button

**Gutenboarding**

* Go to `/new`
* Check the design picker still looks good

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/58429

arthur791004 committed:
Design Picker: Make blank canvas as button

a8ci18n commented on client/signup/steps/design-picker/index.jsx:
```diff
@@ -174,6 +179,21 @@ export default function DesignPickerStep( props ) {
+									pickDesign( designs.find( ( design ) => isBlankCanvasDesign( design ) ) )
+								}
+							>
+								{ translate( 'Use Blank Canvas' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **28** times:
`translate( 'Blank Canvas' )` *ES Score: 8*

p-jackson commented on client/signup/steps/design-picker/style.scss:
```diff
@@ -125,3 +125,24 @@
+		height: 40px;
+		border-radius: 4px;
+		box-shadow: 0 1px 2px rgb( 0 0 0 / 5% );
+		font-size: 0.875rem !important; /* Overwrite style at L46 */
```
Is it possible to use the `design-picker__categories-footer-button` class in some way to override this font style instead?

arthur791004 replied:
As the value to overwrite is `inherit`, I move the `font-size` to the parent style

p-jackson reviewed (commented):
It's working well 👍 

Since we've tried to make the design picker step API driven, I'm wondering if there's a way to keep the new blank canvas button API driven too? What I'm thinking:

- Filter the `theme_picks.includes( 'featured' )` out of `designs` inside the memo hook that does the filter/mapping
- Maybe the memo should return two lists? One with only the `featured` themes and one without?
- Having a loop for the `categoriesFooter` prop, so that it renders a button for each `'featured'`
- The button label uses a `printf` style value of `'Use %s'`

It wouldn't look any different to the user, but since we're trying to allow the theme gallery to drive how the design picker appears, it'd be nice to keep it API driven. E.g. if the blank canvas theme has to change to something else as part of #58431 then it'd be nice to be able to make the change without updating any code.

arthur791004 committed:
Change to use the value of theme picks

arthur791004 commented:
@p-jackson 

> * Filter the theme_picks.includes( 'featured' ) out of designs inside the memo hook that does the filter/mapping
> * Maybe the memo should return two lists? One with only the featured themes and one without?
> * Having a loop for the categoriesFooter prop, so that it renders a button for each 'featured'
> * The button label uses a printf style value of 'Use %s'

I've changed to use theme_picks.includes( 'featured' ) and also added a new property to design named `is_featured_picks` (Or better naming 😅).

Now. the memo would return the object with `designs` and `featuredPicksDesigns` (meaning `is_featured_picks` is true)

Also, I create a new component named `FeaturedPicksButtons`, and then we can use it to render all of “featured” designs as we need this feature p1638178895225900/1637819354.184500-slack-C029SB8JT8S.

arthur791004 committed:
Fix tests

arthur791004 committed:
Use featured picks buttons only when enable categories

p-jackson commented on client/signup/steps/design-picker/index.jsx:
```diff
@@ -174,6 +182,11 @@ export default function DesignPickerStep( props ) {
 					/>
 				}
+				categoriesFooter={
+					useFeaturedPicksButtons && (
```
Do we need to worry about adding a `useFeaturedPicksButtons` prop to the component? The DIFM designs don't include any featured picks, so we don't have to worry about disabling the Blank Canvas button in that flow.

```suggestion
					config.isEnabled( 'signup/design-picker-use-featured-picks-buttons' ) && !! featuredPicksDesigns.length && (
```

arthur791004 replied:
Okay, I think we don't need this prop 👍 Besides, I've already checked `featuredPicksDesigns` inside `FeaturedPicksButtons`, so I think we don't need to check it here?

p-jackson reviewed (approved):
Looks good! Thanks for making the changes.

arthur791004 committed:
Remove useFeaturedPicksButtons prop as it can be configured from designs

arthur791004 committed:
Change to wordpress button

arthur791004 committed:
Make text-align center

arthur791004 commented:
@p-jackson I feel slightly weird to use `theme_picks` for those buttons after consideration. The reason we make the blank canvas button out of the designs is that "blank" is not a design as much as it is customer opting into a different kind of experience in lack of suitable designs to start with (See https://github.com/Automattic/wp-calypso/issues/58429#issuecomment-977998882).

However, if the blank canvas theme has to change to something else as part of #58431, we need to ensure the title is clear enough so that the user can easily know the meaning of `Use %s` button. Or we need to name a “blank” child theme of Blockbase to “Blank Canvas” too and ensure that there is only one blank canvas button below the categories.

p-jackson commented:
That's a fair point: in all of the discussion about this change it hasn't been "let's show a featured one in the sidebar", it's been "let's show _the blank one_ in the sidebar."

The reason why I think it'd all work out fine if the blank theme had to change, and why that taxonomy term would be moved from one to the other, is because it's the same logic that's used to promote Blank Canvas in the `themes/{{ slug }}` view. I like the idea of all the theme galleries responding to the API in the same way, so even if they're not sharing code, at least they're sharing the API nicely. The gallery in the mobile app also sorts Blank Canvas to the top, I assume using the same label.
The `theme_picks` taxonomy doesn't have a great name, and displaying the "recommended picks" as buttons doesn't make a lot of sense. But personally I'd just ignore the name of the slug since Ian basically admitted that they weren't super good.

If you still feel like it's better to special case the Blank Canvas theme specifically, that's fair enough. It can still be API driven in the sense that the button will only appear if the API returns Blank Canvas as an option, and we just filter it out of `designs` by slug.

arthur791004 commented:
Yep, I also like all the idea of the theme galleries responding to the API in the same way. I think we can keep the current approach and see there is any further work after we come up with the better blank canvas.

> The theme_picks taxonomy doesn't have a great name, and displaying the "recommended picks" as buttons doesn't make a lot of sense.

Maybe we should have something like the “blank-canvas” term which might be easier to understand.

jordesign commented:
This works great for me in testing. I may be a little late to raise this - but how do we feel about `Use Blank Canvas` as the button label?

Is 'Blank Canvas' a universal enough term to indicate starting with a blank site? Or is it too much like we're using a theme name, that a new user will have no context to understand.

p-jackson commented:
> Is 'Blank Canvas' a universal enough term to indicate starting with a blank site? Or is it too much like we're using a theme name, that a new user will have no context to understand.

😆 this is very much related to the conversation Arthur and I were having above! Personally I think `Use Blank Canvas` is a fine label. The copy on the screen gives the user context that they're selecting a theme, so they might put two and two together and realise that `Blank Canvas` is also a theme. But if they don't I think that's also fine, the button label works either way imo.

But if we did want to have a more generic "start from scratch" button label, then that'd be another point in @arthur791004's column to special-case the `blank-canvas` theme specifically.

arthur791004 commented:
@jordesign Good concern! I also think the user can easily understand the meaning of the “Blank Canvas”. It's just like we use the “Blank Page” term for choosing the page pattern. Or we can ask @kristastevens's opinion for the copy of “Blank Canvas” 🤔

kristastevens commented:
I do think "Blank Canvas" is clear as a term, fwiw.
